desc/compute/data_index.py

```diff
@@ -45,18 +46,21 @@ def register_compute_fun(
         Coordinate dependency. IE, "rtz" for a function of rho, theta, zeta, or "r" for
         a flux function, etc.
     data : list of str
-        Names of other items in the data index needed to compute qty
+        Names of other items in the data index needed to compute qty.
+    axis_limit_data : list of str
+        Names of other items in the data index needed to compute axis limit of qty.
 
     Notes
     -----
     Should only list *direct* dependencies. The full dependencies will be built
-    recursively at runtime using each quantities direct dependencies.
+    recursively at runtime using each quantity's direct dependencies.
     """
     deps = {
         "params": params,
         "transforms": transforms,
         "profiles": profiles,
         "data": data,
+        "axis_limit_data": [] if axis_limit_data is None else axis_limit_data,
```

Comment:
Using a mutable object as a default parameter can lead to bugs. Because the default parameter of the function stores a reference to some list, if that list is modified, then the default value of the function changes accordingly to the modified list.
We don't plan to ever modify a quantity's `axis_limit_data` list, so you're right that we can use the same empty list for `axis_limit_data` for all quantities that don't specify it in the `compute_funs`. But we use this same `None` to empty list design pattern in other places of the code too.